### ISSUE TITLE
the field 'args' has an invalid value

### ISSUE THREAD

**Issue by dominikkukacka** [dresden-weekly/ansible-network-interfaces#62]
This yml throws an error and I can't really figure out why

```yml
- hosts: server
  remote_user: root
  roles:
    - role: dresden-weekly.network-interfaces
      network_interfaces:
      - device: eth0
        description: private interface
        auto: true
        family: inet
        method: static
        address: 10.0.0.2
        netmask: 255.255.255.192
        allow:
        - hotplug
        post-up:
        - route add -net 10.0.0.0 netmask 255.0.0.0 gw 10.0.0.1
        - route add -net 161.26.0.0 netmask 255.255.0.0 gw 10.0.0.1
```

```bash
fatal: [...]: FAILED! => {"failed": true, "msg": "the field 'args' has an invalid value, which appears to include a variable that is undefined. The error was: 'dict object' has no attribute 'path'\n\nThe error appears to have been in '/private/etc/ansible/roles/dresden-weekly.network-interfaces/tasks/manage_devices.yml': line 8, column 3, but may\nbe elsewhere in the file depending on the exact syntax problem.\n\nThe offending line appears to be:\n\n# depending on change the result stores dest or path\n- name: configured files\n  ^ here\n"}
```

An ideas? Best, Dominik

**Comment by arBmind** [comment]
Strange... I wonder what the structure of the `network_configuration_result` looks for you? Somehow the result structure seems to have changed, or the `default`-chain is somehow broken.

**Comment by dominikkukacka** [comment]
I tried to print it with debug:

```bash
fatal: [...]: FAILED! => {"failed": true, "msg": "the field 'args' has an invalid value, which appears to include a variable that is undefined. The error was: 'network_configuration_result' is undefined\n\nThe error appears to have been in '/Users/dominik/work/.../.../ansible/networking/tasks/main.yml': line 2, column 3, but may\nbe elsewhere in the file depending on the exact syntax problem.\n\nThe offending line appears to be:\n\n---\n- debug:\n  ^ here\n"}
```

**Comment by dominikkukacka** [comment]
I just cloned the `develop` branch to print out the  `network_configuration_result` and there it works!

```json
{
    "msg": {
        "changed": true,
        "msg": "All items completed",
        "results": [
            {
                "_ansible_item_result": true,
                "_ansible_no_log": false,
                "_ansible_notify": [
                    "network restart interface service",
                    "network restart interface command"
                ],
                "changed": true,
                "invocation": {
                    "module_args": {
                        "dest": "/etc/network/interfaces.d/device-eth0",
                        "mode": 420,
                        "src": "device.j2"
                    },
                    "module_name": "template"
                },
                "item": [
                    "eth0",
                    [
                        {
                            "address": "10.91.0.2",
                            "allow": [
                                "hotplug"
                            ],
                            "auto": true,
                            "description": "private interface",
                            "device": "eth0",
                            "family": "inet",
                            "method": "static",
                            "netmask": "255.255.255.192",
                            "post-up": [
                                "route add -net 10.0.0.0 netmask 255.0.0.0 gw 10.91.0.1",
                                "route add -net 161.26.0.0 netmask 255.255.0.0 gw 10.91.0.1"
                            ]
                        }
                    ]
                ]
            }
        ]
    }
}
```

**Comment by arBmind** [comment]
You mean we should create a new release?

**Comment by dominikkukacka** [comment]
I am currently using the develop branch directly. but yes a release would
be better.

Am 25.09.2017 19:43 schrieb "Andreas Reischuck" <notifications@github.com>:

**Comment by arBmind** [comment]
Ok, let's see if we can get a fix for #63 into this release.
